### PR TITLE
[Fixes 955] Blured icon images

### PR DESCRIPTION
**Pull request by DavidQuartz**
This PR implements https://github.com/geonode/geonode-mapstore-client/issues/955#issuecomment-1161849063
Screenshot 2022-06-27 at 09 07 31

Screenshot 2022-06-27 at 09 07 15

ref #955

### CODE REVIEW

File: geonode_mapstore_client/client/js/components/MainErrorView/MainErrorView.jsx [hunk 1]
```diff
@@ -11,14 +11,15 @@ import FaIcon from '@js/components/FaIcon';
 import Message from '@mapstore/framework/components/I18N/Message';
 
 function MainErrorView({
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/components/MainErrorView/MainErrorView.jsx:13]: Let's change the name of this component if it has been used as generic event view for different use case.
MainEventView ?

File: geonode_mapstore_client/client/js/components/MediaViewer/Media.jsx [hunk 2]
```diff
@@ -60,7 +48,28 @@ const mediaDefaultProps = {
...
+                newimageHeight = this.naturalHeight;
+                const container = document.getElementById('ms-container');
+                const containerStyles = window.getComputedStyle(container);
+                if (newimageWidth < parseInt(containerStyles.width, 10) && newimageHeight < parseInt(containerStyles.height, 10)) {
+                    return image.classList.add('natural-image');
+                }
+                return false;
+            };
+        }
+    }, []);
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/components/MediaViewer/Media.jsx:71]: This solution seems applied only on the mount this means it's not applied on resize.
Anyway I think we could use a pure css solution without using js.
I was testing some changes in the classes see this sample

```html
<!DOCTYPE html>
<html>
    <head>
        <style>
           .gn-media-viewer {
                height: calc(100% - 2rem);
                left: 0;
                margin: 1rem;
                position: absolute;
                top: 0;
                width: calc(100% - 2rem);
           }
           .gn-media-viewer .ms-media {
                position: relative;
                width: 100%;
                height: 100%;
            }
            .ms-media-image {
                display: flex;
                align-items: center;
                justify-content: center;
            }
            img {
                position: relative;
                max-width: 100%;
                max-height: 100%;
                object-fit: contain;
                cursor: pointer;
                width: auto;
                height: auto;
            }
        </style>
    </head>
    <body>
        <div class="gn-media-viewer">
            <div class="ms-media ms-media-image">
                <!-- large image -->
                <!-- img src="https://master.demo.geonode.org/documents/1673/link"/-->
                <!-- small image -->
                <img src="https://master.demo.geonode.org/documents/1577/link" />
            </div>
        </div>
    </body>
</html>
```
The main changes are related to apply only max-width and max-height to the img element and ensure that the ms-media-image is a flex that align the item on the center. Could you please test this solution?

**Review by allyoucanmap — CHANGES_REQUESTED** [review]
Please update the @mapstore/project dep to 1.0.24 this should solve the issue with the test, thanks

File: geonode_mapstore_client/client/js/components/MediaViewer/Media.jsx (hunk 2 above)
Comment (DavidQuartz) [reply geonode_mapstore_client/client/js/components/MediaViewer/Media.jsx:71]: It works @allyoucanmap

**Review by allyoucanmap — APPROVED** [review]

**Comment by DavidQuartz** [comment]
@marthamareal please test this so it can be back ported

**Comment by marthamareal** [comment]
@DavidQuartz This can be backported. images are viewed with their dimensions. in the view page

**Pull request merged**